mass_mailing_partner/wizard/partner_merge.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class BasePartnerMergeAutomaticWizard(models.TransientModel):
+    _inherit = "base.partner.merge.automatic.wizard"
+
+    def _merge(self, partner_ids, dst_partner=None, extra_checks=True):
+        if dst_partner:
+            contacts = self.env["mailing.contact"].search(
+                [("partner_id", "in", partner_ids)]
+            )
+            if contacts:
+                contacts = contacts.sorted(
+                    lambda x: 1 if x.partner_id == dst_partner else 0
+                )
+                contacts[1:].unlink()
```

Comment:
You are doing this in the wrong order: removing them first, then next call to `mapped` will fail.